src/backend/src/utils/notifications.utils.ts

```diff
@@ -0,0 +1,69 @@
+import { Team, Task as Prisma_Task, WBS_Element } from '@prisma/client';
+import { UserWithSettings } from './auth.utils';
+import { HttpException } from './errors.utils';
+
+export type UserWithTeams = UserWithSettings & {
+  teamAsHead: Team | null;
+  teamsAsLead: Team[] | null;
+  teamsAsMember: Team[] | null;
+};
+
+export type TaskWithAssignees = Prisma_Task & {
+  assignees: UserWithSettings[] | null;
+  wbsElement: WBS_Element | null;
+};
+
+export const usersToSlackPings = (users: UserWithSettings[]) => {
+  // https://api.slack.com/reference/surfaces/formatting#mentioning-users
+  return users.map((user) => `<@${user.userSettings?.slackId}>`).join(' ');
+};
+
+/**
+ * Gets the team of a task's assignees.
+ * Assumes all assigness share a team
+ * @param users the users of the task
+ * @returns the team assigned to the task
+ */
+export const getTeamFromTaskAssignees = (users: UserWithTeams[]): Team => {
+  const allTeams = users.map((user) => {
+    const teams = [];
+    if (user.teamAsHead) teams.push(user.teamAsHead);
+    if (user.teamsAsLead) teams.push(...user.teamsAsLead);
+    if (user.teamsAsMember) teams.push(...user.teamsAsMember);
+    return teams;
+  });
+
+  // Find common teams across all users
+  const commonTeams = allTeams.reduce((acc, teams, index) => {
+    if (index === 0) {
+      return teams;
+    }
+    return acc.filter((team) => teams.some((t) => t.teamId === team.teamId));
+  }, allTeams[0] || []);
```

Comment:
I see what you're doing here, this commonTeams section might read a little simpler if you do it as two for loops (loop through the teams for the first user and for each one check if it has a match in each other user, returning the team if it has a match in every one). That's essentially what you're doing here but some of this code is a bit janky to make it fit into a reduce call